<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot name="spherorobot_rotor_v1">

  <link name="base_link"> </link>
  
  <link name="sphere_link">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="0.08" />
      <inertia
        ixx="0.00017"
        ixy="0"
        ixz="0"
        iyy="0.00024"
        iyz="0"
        izz="0.00017" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="$(find spherorobot_cpp)/meshes/base_sphere_link.STL" />
      </geometry>
      <material name="">
        <color rgba="0.75 0.75 0.75 1" />
      </material>
    </visual>
    <collision>
    <!-- ${pi/2} -->
      <origin xyz="0 0 0" rpy="0 0 0" /> 
      <geometry>
        <!-- <mesh filename="package://spherorobot_rotor_v1/meshes/base_sphere_link.STL" /> -->
        <!-- <cylinder length="0.22" radius="0.06"/> -->
        <sphere radius="0.06"/>
      </geometry>
    </collision>
  </link>

  <gazebo reference="sphere_link">
    <material>Gazebo/Glass</material>
      <mu1>10.5</mu1>  <!-- Низкое трение для стекла -->
      <mu2>10.5</mu2>
  </gazebo>

  <joint name="base_joint" type="fixed">
    <origin xyz="0 0 0" rpy="0 0 0" />
    <parent link="base_link" />
    <child link="sphere_link" />    
  </joint>

  <link name="pendulum_link">
    <inertial>
      <origin xyz="0 0 -0.022" rpy="0 0 0" />
      <mass value="0.176" />
      <inertia
        ixx="0.00021"
        ixy="0"
        ixz="0"
        iyy="0.00017"
        iyz="0"
        izz="0.00016" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="$(find spherorobot_cpp)/meshes/pendulum_link.STL" />
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 -0.015" rpy="0 0 0" />
      <geometry>
        <!-- <mesh filename="$(find spherorobot_cpp)/meshes/pendulum_link.STL" /> -->
        <box size="0.03 0.03 0.03"/>
      </geometry>
    </collision>
  </link>

  <joint name="pendulum_joint" type="continuous">
    <origin xyz="0 0 0" rpy="0 0 0" />
    <parent link="sphere_link" />
    <child link="pendulum_link" />
    <axis xyz="0 1 0" />
    <dynamics damping="0.1" friction="0.025"/> 
  </joint>

  <link name="rotor_link">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="0.016" />
      <inertia
        ixx="0.0000009"
        ixy="0"
        ixz="0"
        iyy="0.0000009"
        iyz="0"
        izz="0.0000017" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="$(find spherorobot_cpp)/meshes/rotor_link.STL" />
      </geometry>
      <material name="">
        <color rgba="0 0 1 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 -0.0015" rpy="0 0 0" />
      <geometry>
        <!-- <mesh filename="$(find spherorobot_cpp)/meshes/rotor_link.STL" /> -->
        <cylinder length="0.003" radius="0.015"/>
      </geometry>
    </collision>
  </link>

  <joint name="rotor_joint" type="continuous">
    <origin xyz="0 0 -0.045728" rpy="0 0 0" />
    <parent link="pendulum_link" />
    <child link="rotor_link" />
    <axis xyz="0 0 -1" />
    <!-- <dynamics damping="0.1" friction="0.025"/>  -->
    <!-- <axis xyz="0 0 -1" /> -->
  </joint>

  <gazebo>
    <plugin name="sphero" filename="libsphero_plugin.so">
      <ros><namespace>/</namespace></ros>
    </plugin>
  </gazebo>

</robot>